src/pk-shared.c

```diff
@@ -452,3 +452,28 @@ pk_string_replace (GString *string, const gchar *search, const gchar *replace)
 out:
 	return count;
 }
+
+/**
+ * pk_get_cmdline_for_pid:
+ * @dbus: the #PkDbus instance
+ * @pid: the process ID to get the command line for.
+ *
+ * Gets the command line for the given PID.
```

Comment:
Would be worthwhile mentioning that this is racy if the PID is reused while it’s called